hooks/tk-multi-workfiles2/basic/scene_operation.py

```diff
@@ -102,24 +103,92 @@ def execute(
                 # the current session will delete all the stages
                 if parent_action == "open_file":
                     return True
+
                 if alias_api.is_empty_file() and len(alias_api.get_stages()) == 1:
                     alias_api.reset()
                     return True
+
+                # Reset for new file
+                open_in_current_stage = self.parent.engine.open_delete_stages_dialog(
+                    new_file=True
+                )
+
+                if open_in_current_stage == QtGui.QMessageBox.Cancel:
+                    return False
+
+                if open_in_current_stage == QtGui.QMessageBox.No:
+                    stage_name = uuid.uuid4().hex
+                    alias_api.create_stage(stage_name)
                 else:
-                    open_in_current_stage = self.parent.engine.open_delete_stages_dialog(
-                        new_file=True
+                    alias_api.reset()
+
+                return True
+
+            elif operation == "prepare_new":
+                # On prepping a new scene, check if there is an Alias template new file to open. The template new file
+                # will be determined by pipeline step.
+                if not context or not context.step:
+                    return
+
+                template_new_file_settings = self.parent.engine.settings.get(
+                    "new_file_template"
+                )
+                if not template_new_file_settings:
+                    return
+
+                # Get the path to the tempalte new file - this will be relative to the hooks/tk-multi-workfiles2 directory
+                template_path = template_new_file_settings.get(
+                    context.step["name"], template_new_file_settings.get("Default")
+                )
+
+                if isinstance(template_path, dict):
+                    asset_template_path = None
+
+                    # Attempt to get retrieve an asset type specific template file
+                    if context.entity and context.entity["type"] == "Asset":
```

Comment:
So I tried to do this, but no template fields are found when I call `context.as_template_fields`. This might be due to how the template is defined? I updated the corresponding `tk-config-default2` PR:
https://github.com/shotgunsoftware/tk-config-default2/pull/107